### ISSUE TITLE
adapter: extended protocol transaction handling is wrong

### ISSUE THREAD

**Issue by mjibson** [MaterializeInc/materialize#14038]
Our extended protocol should adopt what postgres recently did: optionally commit statements it needs to before a Sync. #13996 has a change that allows reads to be done transactionally before the Sync, but DDLs are executed outside of that transaction, so we're in a kinda weird state.

https://www.postgresql.org/message-id/flat/17434-d9f7a064ce2a88a3%40postgresql.org
https://git.postgresql.org/gitweb/?p=postgresql.git&a=commitdiff&h=f92944137

**Comment by jkosh44** [comment]
Once #15255 is merged, this will be fixed but we'll have the opposite problem. We commit all statements before a Sync instead of just a handful like Postgres.

Should we leave this issue open, open a new issue, or close this and say that our implicit transaction handling is good enough?

**Comment by mjibson** [comment]
Close it! We are now within spec, even if overzealous.
